spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestCompressionSettings.java

```diff
@@ -106,6 +107,13 @@ public static void startSpark() {
     TestCompressionSettings.spark = SparkSession.builder().master("local[2]").getOrCreate();
   }
 
+  @Before
+  public void resetSpecificConfigurations() {
+    spark.conf().unset(COMPRESSION_CODEC);
+    spark.conf().unset(COMPRESSION_LEVEL);
+    spark.conf().unset(COMPRESSION_STRATEGY);
+  }
```

Comment:
`spark.sql.iceberg.compression-level = 1` is not a valid compression level setting for gzip. If we don't unset all these options before running each test, the old configs set by the previous run will be left over and make gzip tests fail.

Reply:
for other reviewers: the same is already being done in the Spark 3.5 version of the test and was added by #11333